Simplenote/Classes/UITableView+Simplenote.swift

```diff
@@ -39,4 +39,14 @@ extension UITableView {
 
         return cell
     }
+
+    /// Returns a Header instance of the specified kind
+    ///
+    func dequeueReusableHeaderFooterView<T: UITableViewHeaderFooterView>(ofType type: T.Type) -> T {
+        guard let header = dequeueReusableHeaderFooterView(withIdentifier: T.reuseIdentifier) as? T else {
+            fatalError()
```

Comment:
😬 Understanding that this shouldn't be reached in production, does it need to be a fatalError?  An Assert would serve the same purpose in development and be production safe. Thoughts?

Reply:
The huge thing there is: what do we return, if the target HeaderFooter cannot be instantiated, right?

Do we instantiate a new `T`? (what if such subclass is expecting `awakeFromNib` to fully initialize?). Do we return a base class instance? (that would probably crash down the road).

Or do we convert the return type into an Optional, and guard let elsewhere?(and thus, eventually rendering an empty view).

Of all the alternatives, crashing if that condition is met (cannot dequeue) kiiinda felt like the most reasonable one, since we'd catch it 100% on Sentry, and it's something that will either never happen, or happen 10/10.

WDYT? (Thanks for opening the conversation, LOVE these topics!!!)